docs/static-general-description.md

```diff
@@ -5,120 +5,145 @@ The format supports the following features
+- Like most other objects, a Method is always defined in the context of a Namespace, but that is often as a result of being within the context of an Interface within a Namespace.
 
-### Error
+### Errors
 
-- An IFEX Error definition is an error condition that can appear while (attempting to) execute a given Method.
-- An error is defined by referencing one or more **Type**s that define the data that shall be communicated when an error occurs.  Enumeration types are common, but it could be any defined type.
-- NOTE!  There can be _more than one_ error, each with its own datatype, defined for a method.
-- It is not specified by IFEX how errors are propagated to the caller of a function - that translation is target language/environment specific.  It may be done using a return or output parameter in one programming language and using **Exception**-handling in another.  Communication protocols may implement their own error-condition side-channels, or other methods.
+- An IFEX Error definition is a condition that can appear while (attempting to) execute a given Method.
+- An error is defined by referencing a **datatype** that define the type of information that shall be communicated when an error occurs.  Enumeration types are common, but it could be any defined type.
+- There can be _more than one_ error, each with its own datatype, defined for a method.  
+- NOTE: Don't mistake this for what most programming environments describe, which is to define a single enumeration, or structured/variant type, where each returned _value_ may indicate a different "error".  What IFEX provides is the possibility of defining multiple different error returns, each with their own name and type (similar to multiple input or output-variables in a function).  Through overlays, that enables efficiently separating business-logic errors (that are known when the interface is defined), from transport errors (that are only known later, when a particular target environment has been selected).  The difference is subtle and of course the generated code might be required to combine the information into a single struct/variant/union type because of limitations, but the difference is that it can now be done at the deployment stage, and not at the interface definition stage.
+
+- It is not specified by IFEX core IDL how errors are propagated to the caller of a function - that translation is target language/environment specific.  It may be done using a return or output parameter in one programming language and using Exception-handling in another.  Network communication protocols may implement their own error-condition side-channels or other methods.
 
 ### Event
 
 - An Event is a time-sensitive communication with _fire-and-forget_ behavior, sent between parts in the distributed system.
 - A IFEX Event can carry multiple pieces of data with it, each of which can use an arbitrary data type.
```

Comment:
Whn you anyway are changing, do a search for all "A IFEX" and replace with "An IFEX"